components/brave_wallet/browser/polkadot/polkadot_substrate_rpc_unittest.cc

```diff
@@ -128,4 +128,253 @@ TEST_F(PolkadotSubstrateRpcUnitTest, GetChainName) {
+        ",\"changes\":[["
+        "\"0x26aa394eea5630e07c48ae0c9558cef7b99d880ec681799c0cf30e8886371da9de"
+        "1e86a9a8c739864cf3cc5ec2bea59fd43593c715fdd31c61141abd04a99fd6822c8558"
+        "854ccde39a5684e7a56da27d\",null]]}]}");
+
+    polkadot_substrate_rpc_->GetAccountBalance(chain_id, pubkey,
+                                               future.GetCallback());
+
+    auto [account_info, error] = future.Take();
+
+    EXPECT_EQ(error, std::nullopt);
+    EXPECT_EQ(account_info->nonce, 0u);
+    EXPECT_EQ(account_info->consumers, 0u);
+    EXPECT_EQ(account_info->providers, 0u);
+    EXPECT_EQ(account_info->sufficients, 0u);
+
+    uint128_t free = 0;
+    EXPECT_TRUE((uint128_t{account_info->data->free->high} << 64 |
+                 account_info->data->free->low) == free);
+
+    uint128_t reserved = 0;
+    EXPECT_TRUE((uint128_t{account_info->data->reserved->high} << 64 |
+                 account_info->data->reserved->low) == reserved);
+
+    uint128_t frozen = 0;
+    EXPECT_TRUE((uint128_t{account_info->data->frozen->high} << 64 |
+                 account_info->data->frozen->low) == frozen);
+
+    uint128_t flags = uint128_t{0x8000000000000000} << 64;
+    EXPECT_TRUE((uint128_t{account_info->data->flags->high} << 64 |
+                 account_info->data->flags->low) == flags);
+  }
+
+  {
+    url_loader_factory_.AddResponse(
+        testnet_url,
+        "{\"id\":1,\"jsonrpc\":\"2.0\",\"result\":[{\"block\":"
+        "\"0x1bcd3e074b91ef25740714dc63671f4a36d2781ff93877ef9ef31b849d1ad69c\""
+        ",\"changes\":[[\"\", \"0x1234\"]]}]}");
+
+    polkadot_substrate_rpc_->GetAccountBalance(chain_id, pubkey,
+                                               future.GetCallback());
+
+    auto [account_info, error] = future.Take();
+    EXPECT_EQ(error, WalletParsingErrorMessage());
+    EXPECT_FALSE(account_info);
+  }
+
+  {
+    url_loader_factory_.AddResponse(
+        testnet_url,
+        "{\"id\":1,\"jsonrpc\":\"2.0\",\"result\":[{\"block\":"
+        "\"0x1bcd3e074b91ef25740714dc63671f4a36d2781ff93877ef9ef31b849d1ad69c\""
+        ",\"changes\":[]}]}");
+
+    polkadot_substrate_rpc_->GetAccountBalance(chain_id, pubkey,
+                                               future.GetCallback());
+
+    auto [account_info, error] = future.Take();
+    EXPECT_EQ(error, WalletParsingErrorMessage());
+    EXPECT_FALSE(account_info);
+  }
+
+  {
+    url_loader_factory_.AddResponse(
+        testnet_url,
+        "{\"id\":1,\"jsonrpc\":\"2.0\",\"result\":[{\"block\":"
+        "\"0x1bcd3e074b91ef25740714dc63671f4a36d2781ff93877ef9ef31b849d1ad69c\""
+        ",\"changes\":[[]]}]}");
+
+    polkadot_substrate_rpc_->GetAccountBalance(chain_id, pubkey,
+                                               future.GetCallback());
+
+    auto [account_info, error] = future.Take();
+    EXPECT_EQ(error, WalletParsingErrorMessage());
+    EXPECT_FALSE(account_info);
+  }
+
+  {
+    // contains invalid hex in account info
```

Comment:
Please follow common comments format - starts with capital and ends with .

Reply:
Done!